src/target/stm32h7.c

```diff
@@ -257,9 +257,20 @@ void stm32h7_configure_wdts(target_s *const target)
 bool stm32h7_probe(target_s *target)
 {
 	const adiv5_access_port_s *const ap = cortex_ap(target);
+	DEBUG_TARGET("%s: AP partno %X, part_id %X\n", __func__, ap->partno, target->part_id);
+
 	/* Use the partno from the AP always to handle the difference between JTAG and SWD */
 	if (ap->partno != ID_STM32H72x && ap->partno != ID_STM32H74x && ap->partno != ID_STM32H7Bx)
 		return false;
+
+	/* By now it's likely this is ST silicon, either H7 or MP15x_CM4 */
```

Comment:
ST silicon is established by the manufacturer code in cortexm.c, so we'd suggest this read more like `/* By now it's established this is likely a H7, but check that it's not a MP15 with a bugged AP part code */` as that's what this is aiming to check.